diff --git a/arch/x86/common/source/IDEDriver.cpp b/arch/x86/common/source/IDEDriver.cpp
index 2bf134a4..cd1af234 100644
--- a/arch/x86/common/source/IDEDriver.cpp
+++ b/arch/x86/common/source/IDEDriver.cpp
@@ -13,6 +13,7 @@ uint32 IDEDriver::doDeviceDetection()
   uint32 jiffies = 0;
   uint16 base_port = 0x1F0;
   uint16 base_regport = 0x3F6;
+  uint8 state = 0;
   uint8 cs = 0;
   uint8 sc;
   uint8 sn;
@@ -80,11 +81,16 @@ uint32 IDEDriver::doDeviceDetection()
       outportbp(base_regport, devCtrl);
 
       jiffies = 0;
-      while (!(inportbp(base_port + 7) & 0x58) && jiffies++ < IO_TIMEOUT)
+      while (!((state = inportbp(base_port + 7)) & 0x58) && jiffies++ < IO_TIMEOUT)
         ArchInterrupts::yieldIfIFSet();
 
       if (jiffies >= IO_TIMEOUT)
         debug(IDE_DRIVER, "doDetection: Still busy after reset!\n ");
+      else if (state & 0x01) // ERR
+      {
+        uint8 error = inportbp(base_port + 1);
+        debug(IDE_DRIVER, "doDetection: Error after reset! ERR %x\n", error);
+      }
       else
       {
         outportbp(base_port + 6, (cs % 2 == 0 ? 0xA0 : 0xB0));
