diff --git a/userspace/tests/tortillas-expect.c b/userspace/tests/tortillas-expect.c
new file mode 100644
index 00000000..c41c22be
--- /dev/null
+++ b/userspace/tests/tortillas-expect.c
@@ -0,0 +1,28 @@
+/*
+---
+category: tortillas
+description: "This test demonstrates tortillas_expect."
+*/
+
+#include "stdio.h"
+#include "assert.h"
+
+#define ITERS 50
+
+int count = 0;
+
+void task()
+{
+  printf("count %d\n", count);
+  count++;
+}
+
+int main()
+{
+  for (int i = 0; i < ITERS; i++)
+    tortillas_expect("count %d\n", i);
+
+  for (int i = 0; i < ITERS; i++)
+    task();
+}
+
