diff --git a/userspace/libc/include/assert.h b/userspace/libc/include/assert.h
index 20e488f4..50e93c6b 100644
--- a/userspace/libc/include/assert.h
+++ b/userspace/libc/include/assert.h
@@ -7,3 +7,8 @@
     printf("Assertion failed: '%s', file %s, function %s, line %d\n", #X, __FILE__, __FUNCTION__, __LINE__); \
  exit(-1); } } while (0)

+/*
+* Wrapper around printf. Everything, that gets printed with this gets
+* prefixed with 'TORTILLAS EXPECT: '.
+**/
+#define tortillas_expect(FMT, ...) printf("TORTILLAS EXPECT: " FMT, __VA_ARGS__);
