diff --git a/common/include/console/debug.h b/common/include/console/debug.h
index df50d37b..22ceb015 100644
--- a/common/include/console/debug.h
+++ b/common/include/console/debug.h
@@ -74,7 +74,7 @@ const size_t FS                 = Ansi_Yellow;
 const size_t RAMFS              = Ansi_White;
 const size_t DENTRY             = Ansi_Blue;
 const size_t INODE              = Ansi_Blue;
-const size_t PATHWALKER         = Ansi_Yellow;
+const size_t PATHWALKER         = Ansi_Yellow | OUTPUT_ENABLED;
 const size_t PSEUDOFS           = Ansi_Yellow;
 const size_t VFSSYSCALL         = Ansi_Yellow;
 const size_t VFS                = Ansi_Yellow | OUTPUT_ENABLED;
